test-data/stdlib-samples/3.2/test/test_genericpath.py

```diff
@@ -23,7 +23,7 @@ def safe_rmdir(dirname: str) -> None:
 
 class GenericTest(unittest.TestCase):
     # The path module to be tested
-    pathmodule = genericpath # type: Any
+    pathmodule = genericpath
```

Comment:
Ok, so taking several things separately:

1) Why is it better to keep `# Any` annotation for the module instead of `# type: ignore` in the one specific location mypy doesn't understand? It seems to me the latter provides a better test case for mypy, as it will validate all other attribute accesses on the module. If this were in my own code base, I would definitely prefer the `# type: ignore` on one special-case line over an `# Any` annotation that turns off all checking for that variable.

2) Nonetheless, an explicit annotation of `Any` type should override the module-ref propagation; that was a bug in this PR, very good catch! Pushing fix and test.

3) I don't understand why an explicit annotation of `types.ModuleType` should prevent propagation of the module-reference information. It doesn't seem consistent with how class references work. To extend the above example slightly:

```
from typing import Type
class A:
    foo = 'bar'
var: Type[A] = A
reveal_type(var.foo)
```

This code type-checks fine, and reveals a type of `builtins.str`. To me, this is perfectly parallel to:

```
import m
import types
mod: types.ModuleType = m
reveal_type(m.a)
```

I would definitely expect the latter code to succeed and reveal type of `builtins.str`, not fail with `Module has no attribute "a"`. Why does giving an explicit type of `types.ModuleType` imply that I want mypy to play dumb about which module it is and what attributes it has?

I've pushed the behavior that makes sense to me for now; will be happy to update once I understand why.